File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums
        self.prefix = []
        self.prefix.append(self.nums[0]) 
        for i in range(1, len(nums)):
            self.prefix.append(self.prefix[-1] + nums[i])
        print(self.prefix)

    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.prefix[right]
        return self.prefix[right] - self.prefix[left - 1]

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {
    int[] prefix;
    public NumArray(int[] nums) {
        this.prefix = new int[nums.length];
        this.prefix[0] = nums[0];
        for(int i = 1 ; i < nums.length; i++){
            this.prefix[i] = this.prefix[i-1] + nums[i];
        }
    }
    
    public int sumRange(int left, int right) {
        if (left == 0)
            return this.prefix[right];
        return this.prefix[right] - this.prefix[left-1];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */